### PR TITLE
build: relay `exitcode` from imagebuildah to registry

### PR DESCRIPTION
**Pull request by flouthoc**
Podman does not relay exit code from buildah instead returns a generic
error code `125`. Following PR allows `podman` to relay exit code from
`imagebuildah` to `registry` as it is.

Closes: https://github.com/containers/podman/issues/12616

### CODE REVIEW

**Comment by flouthoc** [comment]
Lets see if other tests are happy with this.

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Comment by flouthoc** [comment]
> Lets see if other tests are happy with this.

As expected I have to adjust older tests to receive actual error code instead of `125` :\

File: cmd/podman/images/build.go [hunk 1]
```diff
@@ -290,6 +292,13 @@ func build(cmd *cobra.Command, args []string) error {
 
 	report, err := registry.ImageEngine().Build(registry.GetContext(), containerFiles, *apiBuildOpts)
 	if err != nil {
+		exitCode := buildahCLI.ExecErrorCodeGeneric
+		if ee, ok := (errors.Cause(err)).(*exec.ExitError); ok {
+			if w, ok := ee.Sys().(syscall.WaitStatus); ok {
+				exitCode = w.ExitStatus()
+			}
```
Comment (Luap99) [inline cmd/podman/images/build.go:299]: There is no need for this, you can directly call `ee.ExitCode()`

Comment (rhatdan) [reply cmd/podman/images/build.go:299]: Nice.

Comment (flouthoc) [reply cmd/podman/images/build.go]: @Luap99 Fixed in latest commit however I think go versions older than `1.12` does not contains `ExitCode()` but I think we are not going to be using `1.12` so we should be good.

**Pull request merged**